frontend/src/components/NewCardForm/NewCardForm.jsx

```diff
@@ -0,0 +1,155 @@
+import React from "react";
+import { useState } from "react";
+import "./newcardform.css";
+
+export default function NewCardForm({
+  setNewCardFormOpened,
+  handleNewCardSubmission,
+  boardId,
+}) {
+  const [cardTitle, setCardTitle] = useState("");
+  const [cardMessage, setCardMessage] = useState("");
+  const [cardAuthor, setCardAuthor] = useState("");
+  const [gifQuery, setGifQuery] = useState("");
+  const [gifResults, setGifResults] = useState([]);
+  const [selectedGifUrl, setSelectedGifUrl] = useState("");
+
+  function handleClose() {
+    setNewCardFormOpened(false);
+  }
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    //here construct object from stuff user typed and then send it up to parent
+    const newCard = {
+      title: cardTitle,
+      message: cardMessage,
+      author: cardAuthor ? cardAuthor : "Anonymous",
+      boardId: boardId,
+      upvotes: 0,
+      gif: selectedGifUrl,
+    };
+    handleNewCardSubmission(newCard);
+  };
+
+  const handleGifSearch = async () => {
+    if (!gifQuery) return; //if there is no query return cause we cant search
+    const apiKey = import.meta.env.VITE_GIF_KEY;
+    const gifLimit = 6;
+    const url = `https://api.giphy.com/v1/gifs/search?api_key=${apiKey}&q=${encodeURIComponent(
```

Comment:
Move also the api request to the utils, is easy to track the code once you have split and group the code based on the functionality/purpose